cmd/contour/serve.go

```diff
@@ -303,9 +315,25 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	eh.Metrics = metrics
 	eh.CacheHandler.Metrics = metrics
 
+	synced := make([]cache.InformerSynced, 0, len(informers))
+	for _, inf := range informers {
+		synced = append(synced, inf.HasSynced)
+	}
+
 	// step 13. create grpc handler and register with workgroup.
 	g.Add(func(stop <-chan struct{}) error {
 		log := log.WithField("context", "grpc")
+
+		log.Printf("waiting for informer caches")
+		cache.WaitForCacheSync(stop, synced...)
```

Comment:
If `WaitForCacheSync` returns false, it means the stop chan was signalled and we should (log and) bail

https://github.com/kubernetes/client-go/blob/master/tools/cache/shared_informer.go#L211

Reply:
👍